File: frontend/lib/interviewService.ts
import executionApi from "./executionApi";

export interface InterviewAnswer {
  questionIndex: number;
  question: string;
  transcript: string;
  durationSec: number;
}

export interface InterviewResponse {
  id: string;
  workerId: string;
  workerName: string;
  workerEmail: string;
  answersJson: string;
  interviewScore: number;
  scoringReason: string | null;
  status: "SUBMITTED" | "UNDER_REVIEW" | "SCORED";
  submittedAt: string;
  reviewedAt: string | null;
}

export const interviewService = {
  async submitInterview(answers: InterviewAnswer[]): Promise<InterviewResponse> {
    const res = await executionApi.post<InterviewResponse>("/api/interviews", {
      answersJson: JSON.stringify(answers),
    });
    return res.data;
  },

  async getMyInterview(): Promise<InterviewResponse> {
    const res = await executionApi.get<InterviewResponse>("/api/interviews/my");
    return res.data;
  },

  async getAllInterviews(): Promise<InterviewResponse[]> {
    const res = await executionApi.get<InterviewResponse[]>("/api/interviews/all");
    return res.data;
  },

  async scoreInterview(id: string, score: number): Promise<InterviewResponse> {
    const res = await executionApi.patch<InterviewResponse>(`/api/interviews/${id}/score`, { score });
    return res.data;
  },

  async aiScoreInterview(id: string): Promise<InterviewResponse> {
    const res = await executionApi.post<InterviewResponse>(`/api/interviews/${id}/ai-score`);
    return res.data;
  },
};

File: backend/ProjectWorker-Execution-Service/src/main/java/com/example/ProjectWorker_Execution_Service/service/InterviewService.java
package com.example.ProjectWorker_Execution_Service.service;

import com.example.ProjectWorker_Execution_Service.dto.InterviewRequest;
import com.example.ProjectWorker_Execution_Service.dto.InterviewResponse;
import com.example.ProjectWorker_Execution_Service.security.UserPrincipal;

import java.util.List;

public interface InterviewService {

    InterviewResponse submitInterview(InterviewRequest request, UserPrincipal principal);

    InterviewResponse getMyInterview(UserPrincipal principal);

    List<InterviewResponse> getAllInterviews();

    InterviewResponse scoreInterview(String interviewId, double score);

    InterviewResponse aiScoreInterview(String interviewId);
}
